firebase-firestore-ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -34,10 +34,24 @@ val Firebase.firestore
 /** Returns the [FirebaseFirestore] instance of a given [FirebaseApp]. */
 fun Firebase.firestore(app: FirebaseApp): FirebaseFirestore = FirebaseFirestore.getInstance(app)
 
-/** Returns the contents of the document converted to a POJO or null if the document doesn't exist. */
+/**
+ * Returns the contents of the document converted to a POJO or null if the document doesn't exist.
+ *
+ * @param T The Java class to create.
```

Comment:
Quote about `T`: currently `T` is using the default bound of `Any?`. This would allow you to request a value of type `Int`, despite the fact that this can't actually work--the parameter must be a structured type, containing fields.

Java's generic syntax is sufficiently constrained that we can't express this constraint. Is there a way to do so in Kotlin? If not, no worries, but if possible without too much verbosity this would be worthwhile.

Reply:
Afaik it is not possible. There is nothing special about `Int`, `Float` at the type level, they are just subtypes of `Any` just like any other user-defined type.